rpcs3/Emu/Cell/PPUTranslator.cpp

```diff
@@ -904,18 +904,32 @@ void PPUTranslator::VLOGEFP(ppu_opcode_t op)
 
 void PPUTranslator::VMADDFP(ppu_opcode_t op)
 {
+	auto a = get_vr<f32[4]>(op.va);
+	auto b = get_vr<f32[4]>(op.vb);
+	auto c = get_vr<f32[4]>(op.vc);
+
+	// Optimization: Emit only a floating multiply if the addend is zero
+	value_t<u32[4]> const_value = get_vr<u32[4]>(op.vb);
+	if (auto cv = llvm::dyn_cast<llvm::Constant>(const_value.value))
```

Comment:
I believe get_const_vector supports f32[4] directly, so you don't need const_value.